p2panda-net-next/src/actors/iroh/endpoint.rs

```diff
@@ -0,0 +1,183 @@
+
+        let socket_address_v4 = SocketAddrV4::new(config.bind_ip_v4, config.bind_port_v4);
+        let socket_address_v6 = SocketAddrV6::new(config.bind_ip_v6, config.bind_port_v6, 0, 0);
+
+        // Register list of possible "home relays" for this node.
+        let relay_provided = !config.relay_urls.is_empty();
+        let relay_map = iroh::RelayMap::from_iter(config.relay_urls);
+        let relay_mode = iroh::RelayMode::Custom(relay_map);
+
+        // Create and bind the endpoint to the socket.
+        // @TODO: Add static provider to register addresses coming from our discovery mechanism.
+        let endpoint = iroh::Endpoint::builder()
+            .secret_key(from_private_key(args.private_key.clone()))
+            .transport_config(transport_config)
+            .relay_mode(relay_mode)
+            .bind_addr_v4(socket_address_v4)
+            .bind_addr_v6(socket_address_v6)
+            .bind()
+            .await?;
+
+        let accept_handle = {
+            let endpoint = endpoint.clone();
+            tokio::spawn(async move {
+                loop {
+                    let Some(incoming) = endpoint.accept().await else {
+                        break; // Endpoint is closed.
+                    };
+                    myself.send_message(ToIrohEndpoint::Incoming(incoming));
+                }
+            })
+        };
+
+        Ok(IrohState {
+            args,
+            endpoint,
+            protocols: Arc::default(),
+            accept_handle,
+            worker_pool: ThreadLocalActorSpawner::new(),
+        })
+    }
+
+    async fn post_stop(
+        &self,
+        _myself: ActorRef<Self::Msg>,
+        state: &mut Self::State,
+    ) -> Result<(), ActorProcessingErr> {
+        state.endpoint.close().await;
+        state.accept_handle.abort();
+        Ok(())
+    }
+
+    async fn handle(
+        &self,
+        myself: ActorRef<Self::Msg>,
+        message: Self::Msg,
+        state: &mut Self::State,
+    ) -> Result<(), ActorProcessingErr> {
+        match message {
+            ToIrohEndpoint::RegisterProtocol(alpn, protocol_handler) => {
+                let mixed_protocol_id =
+                    hash_protocol_id_with_network_id(&alpn, &state.args.network_id);
+                let mut protocols = state.protocols.write().await;
+                protocols.insert(mixed_protocol_id.to_vec(), protocol_handler);
+            }
+            ToIrohEndpoint::Connect(node_addr, alpn, reply) => {
+                let mixed_protocol_id =
+                    hash_protocol_id_with_network_id(&alpn, &state.args.network_id);
+
+                // This actor will shut down immediately after the connection was established. The
+                // responsibility to handle the connection object is shifted to the caller from
+                // this point on, so using this actor to reason about the state of the connection
+                // is not possible here.
+                IrohConnection::spawn_linked(
```

Comment:
Since there's no supervision relationship with the spawned actor, I believe you should be able to call `Actor::spawn(...)` instead and omit the `myself.into()` argument.

Reply:
I'm a bit confused by the syntax here...we use `Actor::spawn_linked()` everywhere else in the codebase so it seems odd to call `IrohConnection::spawn_linked()` here. But maybe that has something to do with this being a `ThreadLocalActor`? I haven't worked with that type before so I may be missing something. If it's possible to have a unified approach (which matches the rest of the code), that would be best. If not, that's also fine - I just have to read some more to get used to this approach.